proposals/0357-alpenglow_validator_admission_ticket.md

```diff
@@ -0,0 +1,219 @@
+
+## Motivation
+
+Adding more staked validators to a blockchain does come with costs. At the
+very least, the votes and corresponding rewards need to be processed by
+every voting validator. Therefore, every additional staked validator increases
+the number of messages for every validator.
+
+Right now every voting validator pays voting fee on any vote transaction
+included in a block. The voting fee adds up to ~2 SOL per epoch if the
+validator votes most of the time. It is a burden, yet at the same time it's an
+economic barrier to having too many voting validators.
+
+VAT is only a temporary solution to maintain the current economic equilibrium.
+This proposal intentionally strives to keep voting validator protocol costs
+similar to pre-Alpenglow consensus.
+
+## New Terminology
+
+- **Validator Admission Ticket(VAT)**: The 1.6 SOL charged once per epoch to
+every validator eligible to participate in the next epoch
+
+## Different Vote Related Accounts
+
+Before Alpenglow, we have the following accounts related to voting (the
+identity account and vote authority account can be the same):
+
+- Vote account for saving all the vote states and receiving commission
+
+- Identity account for receiving block rewards
+
+- Vote authority account for signing vote transactions
+
+After Alpenglow, we will still have roughly the same accounts serving the same
+purpose:
+
+- Vote account: This account must contain the correct BLS public key
+corresponding to vote authority keypair. It continues to keep all the vote
+credits and all validator identity/authority/commission information updates
+happen here, but it doesn’t contain vote information any more.
+
+- Identity account: This is the account for the 1.6 SOL VAT; it continues
+receiving block rewards and commissions.
+
+- Vote authority account: For signing BLS messages
+
+## Detailed Design
+
+### What is considered a valid vote account
+
+The VAT discussion in SIMD 326 proposes:
+
+- The VAT is a temporary measure before we implement a better mechanism
+
+- There is a strict limit of the 2,000 highest staked validators
+
+- Before being admitted to an epoch, the 1.6 SOL per epoch VAT fee is deducted
+  from each validator and burned
+
+Another requirement (not explicitly mentioned in SIMD 326 but in the Alpenglow
+White Paper) is that Alpenglow uses BLS to aggregate signatures. So a future
+Alpenglow vote account must contain a valid BLS public key deduced from its
+current vote authority ed25519 private key.
+
+In summary, a valid vote account in an Alpenglow epoch must contain:
+
+- a BLS public key
+
+- at least 1.6 SOL VAT fee plus the necessary storage rent amount for a new
+epoch in its corresponding identity account
+
+When the staked validators for a new epoch are calculated, the leader will
+perform the following operations:
+
+- When there are more than 2,000 valid validators, sort all valid vote accounts
```

Comment:
sorting is fine for now with the 2k limit, but is not generally a scalable operation

Reply:
Yup, I think we should have the new better approach long before we hit more validators.